mlflow/store/local_artifact_repo.py

```diff
@@ -4,19 +4,26 @@
 from mlflow.store.artifact_repo import ArtifactRepository
 from mlflow.utils.file_utils import (build_path, exists, mkdir, list_all, get_file_info,
                                      get_relative_path)
+from mlflow.utils.validation import path_not_unique
 
 
 class LocalArtifactRepository(ArtifactRepository):
     """Stores artifacts as files in a local directory."""
 
     def log_artifact(self, local_file, artifact_path=None):
+        if artifact_path and path_not_unique(artifact_path):
+            raise Exception("Artifact path should be relative to artifact_uri "
+                            "and cannot start with '/' or '.'")
```

Comment:
This error message is a bit confusing since we do allow '.'. Maybe we could reword the message, or use the `_bad_path_message` thrown in the validation?